File: backend/app/schemas/ai_analyst.py
from __future__ import annotations

from pydantic import BaseModel
from typing import Optional


class ChatMessage(BaseModel):
    role: str
    content: str


class AttachmentMeta(BaseModel):
    filename: str
    contentType: str
    size: int
    summary: Optional[str] = None


class AiAnalystRequest(BaseModel):
    message: str
    history: list[ChatMessage] = []
    model: str = "claude-sonnet-4-20250514"
    attachments: list[AttachmentMeta] = []


class QuotePayload(BaseModel):
    price: float
    change: float
    changePercent: float
    volume: int = 0


class PricePoint(BaseModel):
    date: str
    open: float
    high: float
    low: float
    close: float
    volume: int


class AnalysisSection(BaseModel):
    key: str
    title: str
    content: str


class AiAnalystResponse(BaseModel):
    reply: str
    ticker: Optional[str] = None
    companyName: Optional[str] = None
    sector: Optional[str] = None
    rating: Optional[str] = None
    confidenceScore: Optional[float] = None
    quote: Optional[QuotePayload] = None
    chart: list[PricePoint] = []
    analysisSections: list[AnalysisSection] = []
    disclaimer: str = ""
    modelUsed: str = ""


class UploadResponse(BaseModel):
    filename: str
    contentType: str
    size: int
    summary: str

File: types/ai-analyst.ts
export interface ChatMessage {
  role: "user" | "assistant";
  content: string;
}

export interface QuotePayload {
  price: number;
  change: number;
  changePercent: number;
  volume: number;
}

export interface AnalysisPricePoint {
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
}

export interface AnalysisSection {
  key: string;
  title: string;
  content: string;
}

export interface AttachmentMeta {
  filename: string;
  contentType: string;
  size: number;
  summary?: string;
}

export interface AiAnalystResponse {
  reply: string;
  ticker?: string;
  companyName?: string;
  sector?: string;
  rating?: string;
  confidenceScore?: number;
  quote?: QuotePayload;
  chart: AnalysisPricePoint[];
  analysisSections: AnalysisSection[];
  disclaimer: string;
  modelUsed?: string;
}

export interface ModelOption {
  id: string;
  label: string;
}

export const ANTHROPIC_MODELS: ModelOption[] = [
  { id: "claude-sonnet-4-20250514", label: "Sonnet 4" },
  { id: "claude-opus-4-20250514", label: "Opus 4" },
  { id: "claude-haiku-4-20250514", label: "Haiku 4" },
];
